md5.c

```diff
@@ -26,45 +26,22 @@
 #include "config.h"
 #include <stddef.h>
 #include <string.h>
+#include <endian.h>
```

Comment:
`endian.h` is [glibc specific](http://man7.org/linux/man-pages/man3/endian.3.html). My system doesn't ship it, although it has a `sys/endian.h`. This doesn't look neither POSIX.2001 nor C99 to me ;) Currently this branch doesn't build on FreeBSD.